radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -58,15 +56,57 @@ impl TransactionResult {
 
 #[derive(Debug, Clone, ScryptoSbor)]
 pub struct CommitResult {
+    pub state_updates: StateDiff,
```

Comment:
After looking at the simulator example use case, I found it easier to consume the summary (rather than the full ancestor chain).

What do you intend to use the full ancestor chain for?

Reply:
* Allowing separating royalty updates from other changes
* Allowing client to aggregate at an internal component level instead of at a global level
* Making Gateway ingestion a lot more efficient by avoiding lots of database lookups for each receipt (which will buy us much more TPS in the Gateway)
* Enabling ejecting events for The Graph or similar tooling with sufficient information to allow sensible indexing from a firehose